src/cg.jl

```diff
@@ -114,40 +118,48 @@ struct CGStateVariables{T,Tx<:AbstractArray{T}}
 end
 
 function cg_iterator!(x, A, b, Pl = Identity();
-    tol = sqrt(eps(real(eltype(b)))),
-    maxiter::Int = size(A, 2),
-    statevars::CGStateVariables = CGStateVariables(zero(x), similar(x), similar(x)),
-    initially_zero::Bool = false
-)
+                      abstol::Real = zero(real(eltype(b))),
+                      reltol::Real = sqrt(eps(real(eltype(b)))),
+                      tol = nothing, # TODO: Deprecations introduced in v0.8
+                      maxiter::Int = size(A, 2),
+                      statevars::CGStateVariables = CGStateVariables(zero(x), similar(x), similar(x)),
+                      initially_zero::Bool = false)
     u = statevars.u
     r = statevars.r
     c = statevars.c
     u .= zero(eltype(x))
     copyto!(r, b)
 
+    # TODO: Deprecations introduced in v0.8
+    if tol !== nothing
+        Base.depwarn("The keyword argument `tol` is deprecated, use `reltol` instead.", :cg_iterator!)
+        reltol = tol
+    end
+
     # Compute r with an MV-product or not.
     if initially_zero
         mv_products = 0
-        c = similar(x)
-        residual = norm(b)
-        reltol = residual * tol # Save one dot product
     else
         mv_products = 1
         mul!(c, A, x)
         r .-= c
-        residual = norm(r)
-        reltol = norm(b) * tol
     end
+    residual = norm(r)
+    # TODO: According to the docs, the code below should use the initial residual
+    #       instead of the norm of the RHS `b` to set the relative tolerance.
+    # See also https://github.com/JuliaMath/IterativeSolvers.jl/pull/244
+    # tolerance = max(reltol * residual, abstol)
+    tolerance = max(reltol * norm(b), abstol)
```

Comment:
Just use the initial residual, it saves another inner product.

Reply:
Okay, thanks, I will do that and change the behavior to be in accordance with the docs. I will have to change some tests, too.

Reply:
I've made this change. Based on this comment, I'll assume that I shall do the same for the other iterative methods.